crates/cheatcodes/src/inspector.rs

```diff
@@ -1228,6 +1228,51 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
         let address = self.allow_cheatcodes_on_create(data, call);
         // If `recordAccountAccesses` has been called, record the create
         if let Some(recorded_account_diffs_stack) = &mut self.recorded_account_diffs_stack {
+            let mut create_accessor = call.caller;
+            let mut create_call_depth = data.journaled_state.depth();
+
+            let mut base_depth = 1;
+            if let Some(prank) = &self.prank {
+                base_depth = prank.depth;
+            } else if let Some(broadcast) = &self.broadcast {
+                base_depth = broadcast.depth;
+            }
+
+            // If the create scheme is Create2 and the depth equals the broadcast/prank/default
+            // depth, then record an additional call to the create2 factory.
+            if data.journaled_state.depth() == base_depth {
```

Comment:
I believe this PR would result in a correct state diff for scripts, but not tests, because of https://github.com/foundry-rs/foundry/issues/5529